Sources/NIO/ByteBuffer-int.swift

```diff
@@ -113,13 +113,19 @@ extension UInt32 {
 
         var n = self
 
+        #if arch(arm) // 32-bit, Raspi/AppleWatch/etc
+            let max = UInt32(Int32.max)
+        #else
+            let max = UInt32.max
+        #endif
```

Comment:
This change violates the doc comment up above, so we should probably change this.

Incidentally, I assume the reason for this change is that `malloc` accepts `Int` on 32-bit platforms, so returning UInt32.max is not a super great plan, so it may be worth changing the doc comment to say something like:

"Returns the next power of two unless that would overflow, in which case UInt32.max (on 64-bit systems) or Int32.max (on 32-bit systems) is returned. The returned value is always safe to be cast to `Int` and passed to `malloc` on all platforms."

Reply:
@helje5 / @Lukasa wait. malloc takes `size_t` which is unsigned, so it should be `CSize` == `UInt` imho. So I don't quite understand this change

Reply:
This was probably to support the `sysMalloc(Int)` issue. Looks like you changed that upstream to `sysMalloc(size_t)`, so it may not be necessary anymore.